### PR TITLE
Implemented the availablity of one payment mode scenario in mileage a…

### PR DESCRIPTION
**Pull request by saitejasFyle**

### CODE REVIEW

**Review by Dimple16 — APPROVED** [review]

**Comment by Dimple16** [comment]
since these changes are already reviewed and approved for normal expense, the mileage and per diem looks good

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -1424,7 +1424,18 @@ export class AddEditExpensePage implements OnInit {
                 })
             )
           ),
-          of(null)
+
+          /*
+          To do: This is just temporary to simulate testing.
+          Replace this section with of(null) and handle the default payment mode logic in the defaultPaymentMode$ when the APIs are available
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1430]: Please dont send testing code in PRs

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 2]
```diff
@@ -528,7 +531,30 @@ export class AddEditMileagePage implements OnInit {
...
+          isMultipleAdvanceEnabled
+        );
+
+        /* 
+          We have to find whether the payment mode account associated with the expense is actually enabled/available for the user.
+          In case if it is not, we have to append the account asociated with the expense to the available payment modes for the user
+        */
+
+        if (etxn?.tx.source_account_id) {
+          return constructedPaymentModes.pipe(
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:545]: This block of code is being repeated on all three pages right? Can you figure out a way to remove this duplication?

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 3]
```diff
@@ -1204,10 +1235,16 @@ export class AddEditMileagePage implements OnInit {
       )
     );
 
+    /*
+      To do: This is just temporary to simulate testing.
+      Replace this section with the actual default payment mode logic once the APIs are available
+    */
+
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:1242]: Same as above

**Review by Julias0 — CHANGES_REQUESTED** [review]
Added comments

**Pull request closed without merging**